orm/src/main/java/jakarta/data/orm/Merge.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.orm;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Lifecycle annotation for repository methods of stateful repositories
+ * which perform merge operations. The {@code Merge} annotation must
+ * not be applied to a method of a stateless repository.
+ * </p>
+ * <p>The {@code Merge} annotation indicates that the annotated repository
+ * method copies the state of one of more unmanaged entities to managed
+ * entities belonging to the current persistence context associated with the
+ * repository. The merge might change the persistent state of the managed
+ * entities, and might cause that state to be inserted or updated in the
+ * database. This typically occurs later, when the persistence context is
+ * flushed.
```

Comment:
This will only make sense if we also define what it means for a persistence context to be flushed or if the users knows Jakarta Persistence and makes assumptions that it is the same thing.